File: src/main/java/com/ploy/bubble_server_v3/domain/auth/presentation/AuthController.java
package com.ploy.bubble_server_v3.domain.auth.presentation;

import com.ploy.bubble_server_v3.common.jwt.dto.TokenResponse;
import com.ploy.bubble_server_v3.domain.auth.presentation.dto.request.LoginRequest;
import com.ploy.bubble_server_v3.domain.auth.presentation.dto.request.QuitRequest;
import com.ploy.bubble_server_v3.domain.auth.presentation.dto.request.SignUpRequest;
import com.ploy.bubble_server_v3.domain.auth.presentation.dto.request.TokenRefreshRequest;
import com.ploy.bubble_server_v3.domain.auth.service.CommandAuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.annotation.security.PermitAll;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import static com.ploy.bubble_server_v3.common.util.AuthenticationUtil.getUserId;

@Tag(name = "인증/인가")
@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/auth")
public class AuthController {

    private final CommandAuthService commandAuthService;

    @Operation(summary = "회원가입")
    @PostMapping("/signup")
    @PermitAll
    public ResponseEntity<Void> signUp(@Valid @RequestBody SignUpRequest req) {
        commandAuthService.signUp(req);
        return ResponseEntity.noContent().build();
    }

    @Operation(summary = "로그인")
    @PostMapping("/login")
    @PermitAll
    public ResponseEntity<TokenResponse> login(@Valid @RequestBody LoginRequest req) {
        return ResponseEntity.ok(commandAuthService.login(req));
    }

    @Operation(summary = "로그아웃")
    @PostMapping("/logout")
    @PreAuthorize("hasAnyRole('USER')")
    public ResponseEntity<Void> logout(@Valid @RequestBody TokenRefreshRequest req) {
        commandAuthService.logout(req);
        return ResponseEntity.noContent().build();
    }

    @Operation(summary = "회원탈퇴")
    @DeleteMapping
    @PreAuthorize("hasAnyRole('USER')")
    public ResponseEntity<Void> quit(@Valid @RequestBody QuitRequest req) {
        commandAuthService.quit(getUserId(),req);
        return ResponseEntity.noContent().build();
    }

    @Operation(summary = "토큰 갱신")
    @PostMapping("refresh")
    @PreAuthorize("hasAnyRole('USER')")
    public ResponseEntity<TokenResponse> refresh(@Valid @RequestBody TokenRefreshRequest req) {
        return ResponseEntity.ok(commandAuthService.refresh(req));
    }
}

File: src/main/kotlin/com/ploy/bubble_server_v2/domain/user/controller/AuthController.kt
package com.ploy.bubble_server_v2.domain.user.controller

import com.ploy.bubble_server_v2.domain.facade.AuthenticationFacade
import com.ploy.bubble_server_v2.common.util.AuthenticationUtil.getMemberId

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@Tag(name = "인증/인가")
@Slf4j
@RestController
@RequiredArgsConstructor
class AuthController {
    private val authenticationFacade: AuthenticationFacade? = null

    @Operation(summary = "회원가입")
    @PostMapping("/signup")
    fun signUpParent(
        @RequestBody req: @Valid SignUpRequest?
    ): ResponseEntity<String> {
        authenticationFacade.signUp(req)
        return ResponseEntity<String>("회원가입이 완료되었습니다.", HttpStatus.OK)
    }

    @Operation(summary = "로그인")
    @PostMapping("/login")
    fun login(@RequestBody req: LoginRequest?): ResponseEntity<LoginResponse> {
        return ResponseEntity.ok<T>(authenticationFacade.login(req))
    }

    @Operation(summary = "로그아웃")
    @PostMapping("/logout")
    fun logout(): ResponseEntity<Void> {
        authenticationFacade.logout(getMemberId())
        return ResponseEntity.noContent().build<Void>()
    }

    @Operation(summary = "토큰 갱신")
    @PostMapping("/refresh-token")
    fun refreshToken(
        @RequestBody tokenRefreshRequest: @Valid TokenRefreshRequest?
    ): ResponseEntity<TokenResponse> {
        return ResponseEntity.ok<T>(authenticationFacade.refreshToken(tokenRefreshRequest))
    }

    @get:GetMapping("/user")
    @get:Operation(summary = "사용자 정보 확인")
    val userInfo: ResponseEntity<UserResponse>
        get() {
            val userInfo: UserResponse = authenticationFacade.getUserInfo(getMemberId())
            return ResponseEntity.ok<UserResponse>(userInfo)
        }

    @Operation(summary = "회원 탈퇴")
    @DeleteMapping("/user")
    fun deleteUser(): ResponseEntity<Void> {
        authenticationFacade.deleteUser(getMemberId())
        return ResponseEntity.noContent().build<Void>()
    }
}